cli/command/utils.go

```diff
@@ -125,3 +126,34 @@ func AddPlatformFlag(flags *pflag.FlagSet, target *string) {
 	flags.SetAnnotation("platform", "version", []string{"1.32"})
 	flags.SetAnnotation("platform", "experimental", nil)
 }
+
+// ValidateOutputPath validates the output paths of the `export` and `save` commands.
+func ValidateOutputPath(path string) error {
+	dir := filepath.Dir(path)
+	if dir != "" && dir != "." {
+		if _, err := os.Stat(dir); os.IsNotExist(err) {
+			return errors.Errorf("invalid output path: directory %q does not exist", dir)
+		}
+	}
+	// check whether `path` points to a regular file
+	// (if the path exists and doesn't point to a directory)
+	if fileInfo, err := os.Stat(path); !os.IsNotExist(err) {
+		if !fileInfo.Mode().IsDir() && !fileInfo.Mode().IsRegular() {
```

Comment:
You can reduce nesting here;

```suggestion
		if fileInfo.Mode().IsDir() || fileInfo.Mode().IsRegular() {
			return nil
		}
		if err := ValidateOutputPathFileMode(fileInfo.Mode()); err != nil {
			return errors.Wrap(err, fmt.Sprintf("invalid output path: %q must be a directory or a regular file", path))
		}		
```